pkg/typesystem/typesystem.go

```diff
@@ -34,6 +34,7 @@ var (
 	ErrReservedKeywords      = errors.New("self and this are reserved keywords")
 	ErrCycle                 = errors.New("an authorization model cannot contain a cycle")
 	ErrNoEntrypoints         = errors.New("no entrypoints defined")
+	ErrNoEntryPointsLoop     = errors.New("no entrypoints defined with potential loop")
```

Comment:
this error message isn't clear to me, also how is it different from `ErrCycle`'s message? in other words, do we need `ErrNoEntryPointsLoop`?

Reply:
updated